### PR TITLE
Added working makefile

### PR DESCRIPTION
**Pull request by Kozova1**
I created a makefile that can replace run.sh for compilation.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Experimental, haven't verified yet
+CXX:=clang++
+CXXFLAGS:=-Wall -Wextra -Isrc/ -std=c++17 -ggdb
+CXXLIBS:=-lSDL2 -lSDL2_ttf
+SRCFILES := $(shell find src/ -type f -name "*.cpp")
+
+.PHONY: build
+build:
```
Comment (Nufflee) [inline Makefile:8]: There should certainly be some sort of a dependency for this target.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Experimental, haven't verified yet
+CXX:=clang++
+CXXFLAGS:=-Wall -Wextra -Isrc/ -std=c++17 -ggdb
+CXXLIBS:=-lSDL2 -lSDL2_ttf
+SRCFILES := $(shell find src/ -type f -name "*.cpp")
```
Comment (Nufflee) [inline Makefile:5]: Can we do this without using shell?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+.PHONY: build
+build:
+	$(CXX) $(CXXFLAGS) $(CXXLIBS) $(SRCFILES) -o build/editor
```
Comment (Nufflee) [inline Makefile:9]: I think the whole point of make is to be able to parallelize compilation and do it incrementally so compiling everything at once probably isn't very useful.

**Review by Nufflee — CHANGES_REQUESTED** [review]

**Comment by Nufflee** [comment]
Thanks for the epic contribution but it will need a little bit more work in order actually be more useful than run.sh.

File: Makefile (hunk 2 above)
Comment (Kozova1) [reply Makefile:5]: `SRCFILES:=$(wildcard src/**/*.cpp)` Might work, but I can't test it at the moment.

**Pull request closed without merging**